src/kktsystem.jl

```diff
@@ -0,0 +1,339 @@
+    kkt.jac .*= kkt.jacobian_scaling # scaling
+    copyto!(kkt.jac_com, kkt.jac_raw)
+end
+
+function compress_hessian!(kkt::AbstractSparseKKTSystem)
+    nothing
+end
+
+function mul!(y::AbstractVector, kkt::AbstractSparseKKTSystem, x::AbstractVector)
+    mul!(y, Symmetric(kkt.aug_com, :L), x)
+end
+
+function jtprod!(y::AbstractVector, kkt::AbstractSparseKKTSystem, x::AbstractVector)
+    mul!(y, kkt.jac_com', x)
+end
+
+get_kkt(kkt::AbstractSparseKKTSystem) = kkt.aug_com
+
+
+#=
+    SparseReducedKKTSystem
+=#
+
+struct SparseReducedKKTSystem{T, MT} <: AbstractSparseKKTSystem{T, MT}
+    hess::StrideOneVector{T}
+    jac::StrideOneVector{T}
+    pr_diag::StrideOneVector{T}
+    du_diag::StrideOneVector{T}
+    # Augmented system
+    aug_raw::SparseMatrixCOO{T,Int32}
+    aug_com::MT
+    aug_csc_map::Union{Nothing, Vector{Int}}
+    # Jacobian
+    jac_raw::SparseMatrixCOO{T,Int32}
+    jac_com::MT
+    jac_csc_map::Union{Nothing, Vector{Int}}
+    # Info
+    ind_ineq::Vector{Int}
+    ind_fixed::Vector{Int}
+    ind_aug_fixed::Vector{Int}
+    jacobian_scaling::Vector{T}
+end
+
+function SparseReducedKKTSystem{T, MT}(
+    n::Int, m::Int, nlb::Int, nub::Int, ind_ineq, ind_fixed,
+    hess_sparsity_I, hess_sparsity_J,
+    jac_sparsity_I, jac_sparsity_J,
+) where {T, MT}
+    n_jac = length(jac_sparsity_I)
+    n_hess = length(hess_sparsity_I)
+
+    aug_vec_length = n+m
+    aug_mat_length = n+m+n_hess+n_jac
+
+    I = Vector{Int32}(undef, aug_mat_length)
+    J = Vector{Int32}(undef, aug_mat_length)
+    V = Vector{T}(undef, aug_mat_length)
+    fill!(V, 0.0)  # Need to initiate V to avoid NaN
+
+    offset = n+n_jac+n_hess+m
+
+    I[1:n] .= 1:n
+    I[n+1:n+n_hess] = hess_sparsity_I
+    I[n+n_hess+1:n+n_hess+n_jac] .= (jac_sparsity_I.+n)
+    I[n+n_hess+n_jac+1:offset] .= (n+1:n+m)
+
+    J[1:n] .= 1:n
+    J[n+1:n+n_hess] = hess_sparsity_J
+    J[n+n_hess+1:n+n_hess+n_jac] .= jac_sparsity_J
+    J[n+n_hess+n_jac+1:offset] .= (n+1:n+m)
+
+    pr_diag = view(V, 1:n)
+    du_diag = view(V, n_jac+n_hess+n+1:n_jac+n_hess+n+m)
+
+    hess = view(V, n+1:n+n_hess)
+    jac = view(V, n_hess+n+1:n_hess+n+n_jac)
+
+    aug_raw = SparseMatrixCOO(aug_vec_length,aug_vec_length,I,J,V)
+    jac_raw = SparseMatrixCOO(m,n,jac_sparsity_I,jac_sparsity_J,jac)
+
+    aug_com = MT(aug_raw)
+    jac_com = MT(jac_raw)
+
+    aug_csc_map = get_mapping(aug_com, aug_raw)
+    jac_csc_map = get_mapping(jac_com, jac_raw)
+
+    ind_aug_fixed = if isa(aug_com, SparseMatrixCSC)
+        _get_fixed_variable_index(aug_com, ind_fixed)
+    else
+        zeros(Int, 0)
+    end
+    jac_scaling = ones(T, n_jac)
+
+    return SparseReducedKKTSystem{T, MT}(
+        hess, jac, pr_diag, du_diag,
+        aug_raw, aug_com, aug_csc_map,
+        jac_raw, jac_com, jac_csc_map,
+        ind_ineq, ind_fixed, ind_aug_fixed, jac_scaling,
+    )
+end
+
+
+#=
+    SparseAugmentedKKTSystem
+=#
+
+struct SparseAugmentedKKTSystem{T, MT} <: AbstractSparseKKTSystem{T, MT}
```

Comment:
I split the `SparseAugmentedKKTSystem` from `SparseReducedKKTSystem` for clarity purpose. Let me know if you think it would be better to merge these two together